src/sentry/plugins/providers/integration_repository.py

```diff
@@ -27,14 +27,43 @@ def __init__(self, id):
 
     def dispatch(self, request, organization, **kwargs):
         try:
+            # get data via API (so we know it's up to date)
             config = self.get_repository_data(organization, request.DATA)
+
+        except Exception as e:
+            return self.handle_api_error(e)
+
+        else:
+            # check to see if we already have this repository, and update any
+            # data if applicable
+            repo = Repository.objects.filter(
```

Comment:
A thought here is why use filter when you could use `get`? Was it to use `.first()`? See my comment on `first` below. `get()` would be fine here.

Reply:
I had it as a `.get()`, but all my tests crashed. :-(

(That's because it turns out `.get()` doesn't return `None` if no record is found, it raises a `DoesNotExist` error - see https://docs.djangoproject.com/en/2.1/ref/models/querysets/#get . Intuitively I was hoping that a framework written in python would implement `.get()` basically the same way python itself does, but... )